crates/uv-workspace/src/workspace.rs

```diff
@@ -1074,6 +1075,7 @@ async fn find_workspace(
             // Only walk up the given directory, if any.
             options
                 .stop_discovery_at
+                .and_then(Path::parent)
                 .map(|stop_discovery_at| stop_discovery_at != *path)
                 .unwrap_or(true)
         })
```

Comment:
This code isn't used in production, but this updates the behavior in tandem with copying the implementation for `PythonVersionFile::discover` to solve a common footgun where you pass the project root as the place to stop discovery and discovery stopped _before_ looking in that directory.